lib/fileio/commands/baseImportCommand.cpp

```diff
@@ -0,0 +1,252 @@
+
+    syntax.enableQuery(false);
+    syntax.enableEdit(false);
+
+    return syntax;
+}
+
+/* static */
+void*
+MayaUSDImportCommand::creator()
+{
+    return new MayaUSDImportCommand();
+}
+
+/* virtual */
+std::unique_ptr<UsdMaya_ReadJob> MayaUSDImportCommand::initializeReadJob(const MayaUsd::ImportData & data, 
+        const UsdMayaJobImportArgs & args)
+{
+    return std::unique_ptr<UsdMaya_ReadJob>(new UsdMaya_ReadJob(data, args));
+}
+
+/* virtual */
+MStatus
+MayaUSDImportCommand::doIt(const MArgList & args)
+{
+    MStatus status;
+
+    MArgDatabase argData(syntax(), args, &status);
+
+    // Check that all flags were valid
+    if (status != MS::kSuccess) {
+        return status;
+    }
+
+    // Get dictionary values.
+    const VtDictionary userArgs =
+        UsdMayaUtil::GetDictionaryFromArgDatabase(
+            argData,
+            UsdMayaJobImportArgs::GetDefaultDictionary());
+
+    std::string mFileName;
+    if (argData.isFlagSet(kFileFlag)) {
+        // Get the value
+        MString tmpVal;
+        argData.getFlagArgument(kFileFlag, 0, tmpVal);
+        mFileName = tmpVal.asChar();
+
+        // Use the usd resolver for validation (but save the unresolved)
+        if (ArGetResolver().Resolve(mFileName).empty()) {
+            TF_RUNTIME_ERROR(
+                    "File '%s' does not exist, or could not be resolved. "
+                    "Exiting.",
+                    mFileName.c_str());
+            return MS::kFailure;
+        }
+
+        TF_STATUS("Importing '%s'", mFileName.c_str());
+    }
+
+    if (mFileName.empty()) {
+        TF_RUNTIME_ERROR("Empty file specified. Exiting.");
+        return MS::kFailure;
+    }
+
+    std::string mPrimPath;
+    if (argData.isFlagSet(kPrimPathFlag))
+    {
+        // Get the value
+        MString tmpVal;
+        argData.getFlagArgument(kPrimPathFlag, 0, tmpVal);
+        mPrimPath = tmpVal.asChar();
+    }
+
+    // Add variant (variantSet, variant).  Multi-use
+    SdfVariantSelectionMap mVariants;
+    for (unsigned int i=0; i < argData.numberOfFlagUses(kVariantFlag); ++i)
+    {
+        MArgList tmpArgList;
+        status = argData.getFlagArgumentList(kVariantFlag, i, tmpArgList);
+        // Get the value
+        MString tmpKey = tmpArgList.asString(0, &status);
+        MString tmpVal = tmpArgList.asString(1, &status);
+        mVariants.insert( std::pair<std::string, std::string>(tmpKey.asChar(), tmpVal.asChar()) );
+    }
+
+    bool readAnimData = false;
+    if (argData.isFlagSet(kReadAnimDataFlag)) {
+        argData.getFlagArgument(kReadAnimDataFlag, 0, readAnimData);
+    }
+
+    GfInterval timeInterval;
+    if (readAnimData) {
+        if (argData.isFlagSet(kFrameRangeFlag)) {
+            double startTime = 1.0;
+            double endTime = 1.0;
+            argData.getFlagArgument(kFrameRangeFlag, 0, startTime);
+            argData.getFlagArgument(kFrameRangeFlag, 1, endTime);
+            timeInterval = GfInterval(startTime, endTime);

```

Comment:
I'm assuming the user could never accidentally specify these the wrong way around?